backend/src/models/reports/Tab_34_rpt_PA_StatusDashboard.js

```diff
@@ -1,17 +1,55 @@
 const dbConnection = require("@database/databaseConnection");
 const { knex } = dbConnection();
+const _ = require("lodash");
+
+// filter raw query results by the value of the portfolios argument.
+const portfolioFilter = (portfolios) => {
+  if (!portfolios) return `WHERE  fy.is_current = true OR p.project_status = 'Active'`;
+
+  return `WHERE  (fy.is_current = true 
+    OR p.project_status = 'Active')
+    AND (po.id IN (${_.castArray(portfolios).join(",")}))`;
```

Comment:
@ASpiteri-BCGov and I found a way to use Knex' where() function for these filters which might make future report queries a little easier but it doesn't look like he's committed it yet so I can't point to how it was done.

I think we should try to use the second parameter of Knex.raw() here to bind the portfolios array instead of putting it directly in the query as text since I think this could be open to SQL injection.

Reply:
good point. fixed.

also handled the edge case where no portfolio parameter is passed in.

would appreciate a cleaner implementation if you can see one!

Reply:
Looks like this will probably work fine. Just wondering if the solution from @ASpiteri-BCGov's last PR can be used here too to simplify it even more.

Reply:
I think it can. I'm going to try it now. The only complication here might arise from the `WITH` clause and how it creates that temporary table q. 

I think I can just treat the whole query as a unit, however.